src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RazorCodeActionContext.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Microsoft.AspNetCore.Razor.Language;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Refactoring
+{
+    struct RazorCodeActionContext
```

Comment:
This being a struct doesn't save you much given the frequency in which it's created. I'd just make it a class.
```suggestion
    internal sealed class RazorCodeActionContext
```

Reply:
It provides non-nullability, fwiw. Any thoughts?

Reply:
It does give you non-nullability but is that what you'd really want? Aka wouldn't you like to know taht a null code action context came your way instead of trying to figure out why a null ref on one of its parameters or worse, why a source  location of span 0 is making its way through the system?